### PR TITLE
fix: skip toast for already-confirmed transactions cp-13.26.0

### PR DESCRIPTION
**Pull request by n3ps**
## **Description**

When a transaction is seen for the first time, skips onSuccess/onFailure toast for already-confirmed statuses

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: https://consensys.slack.com/archives/C0AQ7U8QAH3/p1775462653284529

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> **Medium Risk**
> Changes transaction lifecycle notifications, which can affect user-facing toasts and edge-case state transitions. Risk is limited in scope to `useTransactionLifecycle` and is covered by updated tests.
> 
> **Overview**
> Updates `useTransactionLifecycle` to **only** trigger `onPending` for newly discovered transactions, and to **skip `onSuccess`/`onFailure`** when a transaction is first seen already in a terminal state.
> 
> Refines tests to remove the unapproved→pending transition case, add coverage for “first appears confirmed” (no toast), and adjust the multi-transaction case to expect only success callbacks when both pending txs confirm.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 365140cf76775f9e61818a63bffda012e0a61a69. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: ui/hooks/useTransactionLifecycle.test.ts [hunk 1]
```diff
@@ -40,21 +40,6 @@ describe('useTransactionLifecycle', () => {
     expect(onPending).toHaveBeenCalledWith(pending('1'));
   });
 
-  it('fires onPending when a known tx transitions into pending (unapproved → submitted)', () => {
```
[the remark this answers is not in the document]
Comment (n3ps) [reply ui/hooks/useTransactionLifecycle.test.ts:43]: updated

File: ui/hooks/useTransactionLifecycle.test.ts [hunk 2]
```diff
@@ -118,14 +99,16 @@ describe('useTransactionLifecycle', () => {
     const onSuccess = jest.fn();
     const { rerender } = renderHook(
       ({ txs }) => useTransactionLifecycle(txs, { onPending, onSuccess }),
-      { initialProps: { txs: [pending('1'), approved('2')] } },
+      { initialProps: { txs: [pending('1'), pending('2')] } },
```
[the remark this answers is not in the document]
Comment (n3ps) [reply ui/hooks/useTransactionLifecycle.test.ts:102]: Updated

File: ui/hooks/useTransactionLifecycle.ts [hunk 3]
```diff
@@ -40,24 +40,15 @@ export function useTransactionLifecycle<
       if (!previous) {
         if (isNowPending) {
           handlers.onPending?.(tx);
-        } else if (isNowSuccess) {
```
Comment (n3ps) [inline ui/hooks/useTransactionLifecycle.ts:43]: Don't call these handlers on newly seen and already-confirmed transactions

**Review by NidhiKJha — APPROVED** [review]

**Review by ameliejyc — APPROVED** [review]
Tested and lgtm
